lib/awful/mouse/init.lua

```diff
@@ -297,8 +297,8 @@ function mouse.object.set_current_client() end
 
 function mouse.object.get_current_wibox()
     local obj = capi.mouse.object_under_pointer()
-    if type(obj) == "drawin" then
-        return obj
+    if type(obj) == "drawin" and obj.wibox then
+        return obj.wibox()
```

Comment:
I can't do that. As we talked about when I implemented that system, the only way to avoid leaks in 5.1 is to use upvals to store the properties. Given this is exactly this type of leaks, I had no choice but to do it like this (at least of the solutions I tried). Otherwise, the leaks test would fail, much, much more often (maybe all the time for 5.1 and the vast majority of the time for luajit, my test pool is too small to confirm).